### PR TITLE
Add Deepdive 2

### PR DESCRIPTION
**Pull request by EvanSimpson**
Express routing & middleware

### CODE REVIEW

File: deepdives/02-middleware/Readme.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
+# Deep Dive 2: Express Routing and Middleware
+
+This document is largely going to be a retelling of the Express [documentation for middleware](http://expressjs.com/guide/using-middleware.html), so feel free to read through that page first and then come back here for a more in-depth explanation.
+
+## What is middlewware?
```
Comment (benkahle) [inline deepdives/02-middleware/Readme.md]: ww

File: deepdives/02-middleware/Readme.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+>
+> - Execute any code.
+> - Make changes to the request and the response objects.
+> - End the request-response cycle.
+> - Call the next middleware in the stack.
+
+*Middleware is a function.*
+It's so simple, and at the same time very powerful.
+All of the middleware you `app.use()`?
+Just a function.
```
Comment (benkahle) [inline deepdives/02-middleware/Readme.md]: Very stream of conscious-y

File: deepdives/02-middleware/Readme.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+The application routes you specify in `app.get()` and `app.post()` are just functions, and therefore middleware!
+This leads us to our next commonly missed point by node beginners:
+
+## Your Application is an Array of middleware
+We've touched on this point before in Olin.js, but only briefly, and it certainly bears repeating.
+**Your application is just a list of functions being executed sequentially.**
+
+<!-- NOTE: The following links are totally liable to
```
Comment (benkahle) [inline deepdives/02-middleware/Readme.md:29]: Is this intended to actually be hidden from the reader?

File: deepdives/02-middleware/example/middleware.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+/*
+  Example Authentication Middleware.
+  This is meant to serve as a simple example of how
+  some middleware can be structured.
+  By no means should this middleware ever be used
```
Comment (benkahle) [inline deepdives/02-middleware/example/middleware.js]: Why not? If you're going to say this I would specify why.

**Comment by benkahle** [comment]
:+1:

File: deepdives/02-middleware/Readme.md (hunk 3 above)
Comment (EvanSimpson) [inline deepdives/02-middleware/Readme.md:29]: Yeah.

File: deepdives/02-middleware/example/middleware.js (hunk 4 above)
Comment (EvanSimpson) [inline deepdives/02-middleware/example/middleware.js]: No error handling, doesn't really do much of anything, passport is a better alternative.

Maybe I'll just recommend Passport as an alternative.

**Pull request merged**